packages/core/_src/stm/TMap/operations/put.ts

```diff
@@ -0,0 +1,82 @@
+import type { Journal } from "@effect/core/stm/STM/Journal"
+import { concreteTArray, InternalTArray } from "@effect/core/stm/TArray/operations/_internal/InternalTArray"
+import { TMap } from "@effect/core/stm/TMap/definition"
+import { concreteTMap } from "@effect/core/stm/TMap/operations/_internal/InternalTMap"
+
+const LOAD_FACTOR = .75 as const
+
+/**
+ * Stores new binding into the map.
+ *
+ * @tsplus fluent ets/TMap put
+ */
+export function put_<K, V>(self: TMap<K, V>, k: K, v: V): USTM<void> {
+  function resize(journal: Journal, buckets: TArray<List<Tuple<[K, V]>>>): void {
+    concreteTArray(buckets)
+    const capacity = buckets.chunk.length
+    const newCapacity = capacity << 1
+    const newBuckets: Array<List<Tuple<[K, V]>>> = []
+    let i = 0
+
+    while (i < capacity) {
+      const pairs = buckets.chunk.unsafeGet(i)!.unsafeGet(journal)
+
+      for (const pair of pairs) {
+        const idx = TMap.indexOf(pair.get(0), newCapacity)
+
+        newBuckets[idx] = List.cons(pair, newBuckets[idx] as List<Tuple<[K, V]>>).asList()
```

Comment:
I don't think the call to `asList` is necessary here. And you can avoid the type-cast by just performing a non-null assertion: `newBuckets[idx]!`.